src/node.cc

```diff
@@ -1110,6 +1111,57 @@ bool ShouldAbortOnUncaughtException(Isolate* isolate) {
 }
 
 
+void PromiseHook(PromiseHookType type,
+                 Local<Promise> promise,
+                 Local<Value> parent) {
```

Comment:
The domain enter/exit callbacks don’t do much, they basically just push or pop an object to an array.